text/0227-separate-semantic-conventions.md

```diff
@@ -0,0 +1,156 @@
+    to live. This directory will be hosted at
+    `https://opentelemetry.io/semantic_conventions/schemas/`
+
+Existing semantic conventions in the specification would be marked as
+deprecated, with documentation denoting the move.
+
+Additionally, if the semantic conventions eventually move to domain-specific
+director structure (e.g. `docs/{domain}/README.md`, with trace, metrics, events
+in the same file), then this can be refactored in the new repository, preserving
+git history.
+
+There will also be the following exceptions in the specification:
+
+- Semantic conventions used to implement API/SDK details will be fully specified
+  and will not be allowed to change in the Semantic Convention directory.
+  - Error/Exception handling will remain in the specification.
+  - SDK configuration interaction w/ semantic convention will remain in the
+    specification. Specifically `service.name`.
+- The specification may elevate some semantic conventions as necessary for
+  compatibility requirements, e.g. `service.instance.id` and
+  [Prometheus Compatibility](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md).
+
+These exceptions exist because:
+
+- Stable portions of the specification already rely on these conventions
+- These conventions are required to implement an SDK today.
+
+As such, the Specification should define the absolute minimum of reserved or
+required attribute names and their interaction with the SDK.
+
+## Internal details
+
+The following process would be used to ensure semantic conventions are
+seamlessly moved to their new location.
+
+- A new repository `open-telemetry/semantic-conventions` will be constructed with
+  the proposed format and necessary `Makefile` / tooling.
+- A moratorium will be placed on Semantic Convention PRs to the specififcation
+  repository. (Caveat that PRs related to this proposal would be allowed).
+- All semantic conventions in the Specification will be copied to the new
+  repository.
+- Semantic conventions in the Specification will be marked as deprecated with
+  links to the new location.
+- The Specification will be updated to require "special" conventions, like
+  `service.name` and configuration interaction.
+- Instrumentation authors will update their `SchemaURL` to
+   `https://opentelemetry.io/semantic_conventions/schemas/{semconv_version}`
+   from
+   `https://opentelemetry.io/schemas/{spec_version}`
+
+## Trade-offs and mitigations
+
+This proposal has a few drawbacks:
+
+- The semantic conventions will no longer be easily referencable form the specification.
+  - This is actually a benefit. We can ensure isolation of convention from
+    specification and require the Specification to use firm language for
+    attributes it requires, like `service.name`.
+  - We will provide x-links from existing location to the new location.
+- Semantic Convention version will no longer match the specification version.
```

Comment:
>- Semantic Convention version will no longer match the specification version.

I think this is fine, but IMO the next version of conventions and of SchemaURL should be v1.20.0 (or whatever is the version of the spec after the split happens).

Reply:
How would deal with the confusion where people would go to the specification version to look for semconv?  

SchemaURL does NOT denote where the definition lies, it only tracks changes.   Right now, the community is tuned into looking at the specification for the current URL.

My suggestion of a new URL is to help aide with training folks to look in a new place for the new version and NOT expect the specification to hold semconv going forward.

Reply:
>How would deal with the confusion where people would go to the specification version to look for semconv?

We are planning to plaster the markdown in spec repo with "WARNING: DEPRECATED/MOVED TO FOO!!!" everywhere, right? We have to do it regardless since the community is tuned to go to the spec repo anyway. I am not sure how keeping the current Schema URL family makes it worse, especially since the Schema URL does not suggest in any way that the semconvs or schema files must be found in the spec repo.

>My suggestion of a new URL is to help aide with training folks to look in a new place for the new version and NOT expect the specification to hold semconv going forward.

I think this is solvable by warning messages in the markdown and links to the new repo. I think we can go a step further and as soon as the transition is complete delete the main content in the spec repo's markdown files and  only keep the warning and links to the new repo. There is no way to be confused when you land in the spec repo and see that.

Reply:
We chatted about this in person. I've updated the PR as follows:

1. we will do the process you suggest.
2. We will NOT change the schema url location
3. We WILL change the schema url version on our first release to denote moving away from schema version.

Reply:
>We WILL change the schema url version on our first release to denote moving away from schema version.

Didi you mean "to denote moving away from spec version"?